homeassistant/components/notify/twitter.py

```diff
@@ -51,17 +54,96 @@ def __init__(self, consumer_key, consumer_secret, access_token_key,
                               access_token_secret)
 
     def send_message(self, message="", **kwargs):
-        """Tweet a message."""
+        """Tweet a message, optionally with media"""
+        media = kwargs.get('media', None)
+        media_id = self.upload_media(media)
+
         if self.user:
-            resp = self.api.request(
-                'direct_messages/new', {'text': message, 'user': self.user})
+            resp = self.api.request('direct_messages/new',
+                                    {'text': message, 'user': self.user,
+                                     'media_ids': media_id})
         else:
-            resp = self.api.request('statuses/update', {'status': message})
+            resp = self.api.request('statuses/update',
+                                    {'status': message, 'media_ids': media_id})
 
         if resp.status_code != 200:
-            import json
-            obj = json.loads(resp.text)
-            error_message = obj['errors'][0]['message']
-            error_code = obj['errors'][0]['code']
-            _LOGGER.error("Error %s : %s (Code %s)", resp.status_code,
-                          error_message, error_code)
+            self.log_error_resp(resp)
+
+    def upload_media(self, media_path=None):
+        """Upload media."""
+        if media_path:
```

Comment:
Shouldn't this method not be called if `media_path` is not set ? 

Also, prefer a guard clause:

```python
if not media_path:
    return

media_type, _ = mimetipes.guess_type(…)
```